frodo.js

```diff
@@ -1,48 +1,41 @@
 /**
```

Comment:
SCRIPT-This function might benefit from using async/await instead of callbacks for better readability.